audiorecorder/src/test/java/org/odk/collect/audiorecorder/recording/AudioRecorderActivityTest.kt

```diff
@@ -15,81 +14,104 @@ import org.junit.Before
 import org.junit.Test
 import org.junit.runner.RunWith
 import org.odk.collect.audiorecorder.AudioRecorderDependencyModule
-import org.odk.collect.audiorecorder.overrideDependencies
+import org.odk.collect.audiorecorder.TestApplication
 import org.odk.collect.audiorecorder.recorder.Recorder
+import org.odk.collect.audiorecorder.setupDependencies
+import org.robolectric.Shadows.shadowOf
 import java.io.File
 
 @RunWith(AndroidJUnit4::class)
 class AudioRecorderActivityTest {
 
     private val fakeRecorder = FakeRecorder()
+    private val recordingRepository = RecordingRepository()
 
     @Before
     fun setup() {
-        getApplicationContext<Application>().overrideDependencies(
+        getApplicationContext<TestApplication>().setupDependencies(
             object : AudioRecorderDependencyModule() {
                 override fun providesRecorder(application: Application): Recorder {
                     return fakeRecorder
                 }
+
+                override fun providesRecordingRepository(): RecordingRepository {
+                    return recordingRepository
+                }
             }
         )
     }
 
     @Test
-    fun launching_startsRecording() {
-        launchActivity<AudioRecorderActivity>()
-        assertThat(fakeRecorder.isRecording(), equalTo(true))
+    fun launching_startsRecordingService_withStart() {
```

Comment:
I thought about introducing Espresso tests which mean we could just test the whole flow without having to separate the bulk of testing out between the Activity and the Service. That would become redundant if then got rid of the Activity and exposed the ViewModel as the main interface of the module though so I think for the moment I'll stick with this structure and see where we end up